main.cpp

```diff
@@ -92,18 +92,111 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add( float lhs, float rhs )
+    {
+        float x = lhs + rhs;
+        return x;
+    }
+    
+    float subtract( float lhs, float rhs )
+    {
+        float x = lhs - rhs;
+        return x;
+    }
+    float multiply( float lhs, float rhs )
+    {
+        float x = lhs * rhs;
+        return x;
+    }
+    float divide( float lhs, float rhs )
+    {
+        if (rhs == 0.f) 
+        {
+            std::cout << std::endl;
+            std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        float x = lhs / rhs;
+        return x;
+    } 
+};
+
+struct DoubleType
+{
+    double add( double lhs, double rhs )
+    {
+        double x = lhs + rhs;
+        return x;
+    }
+    double subtract( double lhs, double rhs )
+    {
+        double x = lhs - rhs;
+        return x;
+    }
+    double multiply( double lhs, double rhs )
+    {
+        double x = lhs * rhs;
+        return x;
+    }
+    double divide( double lhs, double rhs )
+    {
+        if (rhs == 0.0) 
+        {
+            std::cout << std::endl;
+            std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        double x = lhs / rhs;
+        return x;
+    }
+};
+
+struct IntType
+{
+    int add( int lhs, int rhs )
+    {
+        int x = lhs + rhs;
+        return x;
+    }
+    int subtract( int lhs, int rhs )
+    {
+        int x = lhs - rhs;
+        return x;
+    }
+    int multiply( int lhs, int rhs )
+    {
+        int x = lhs * rhs;
+        return x;
+    }
+    int divide( int lhs, int rhs )
+    {
+        if (rhs == 0.0) 
+        {
+            std::cout << "error, integer division by zero will crash the program!" << std::endl;
+            std::cout << "returning lhs" << std::endl;
+            return lhs;
+        }
+        else 
+        {
+        int x = lhs / rhs;
+        return x;
+        }
```

Comment:
no need for the temporary `x`.
simply `return lhs / rhs;`

If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise